R/CompDb-methods.R

```diff
@@ -193,4 +193,18 @@ setMethod("deleteSpectra", signature(object = "CompDb"),
                   stop("'object' does not contain msms spectra")
               }
               object
-          })
\ No newline at end of file
+          })
+
+#' @inherit MetaboCoreUtils::mass2mz
+#' 
+#' @importFrom MetaboCoreUtils mass2mz
+#' 
+#' @export
+#' 
+#' @rdname CompDb
+setMethod("mass2mz", signature = c("CompDb"),
+          function(x, adduct){
+              exact_mass <- compounds(x)$exactmass
```

Comment:
Note: each `compounds` call will require an SQL call to the database. I think it would be more efficient to extract all relevant information in one `compounds` call.

I would maybe even suggest the following:

```r
setMethod("mass2mz", signature = c("CompDb"), function(x, adduct = "[M+H]+", name = "compound_id") {
    cmps <- compounds(x, c(name, "exactmass"))
    res <- mass2mz(cmps$exactmass, adduct = adduct)
    rownames(res) <- cmps[[name]]
    res
})
cmps <- compounds(x, c("compound_id", "exactmass", "formula"))
```

this is obviously open for discussion. The default option `mass2mz` with `name = "compound_id"` would return m/z values for all compounds in the database (even if some masses/formulas would be duplicated). But the user would have the m/z for the specified adducts for each compound (and can go back to the associated compound using the `"compound_id"`/name of the value). 
Alternatively, with `name = "formula"` you would get m/z values for **unique** chemical formulas from the database. Does this make sense or is it too complicated?

Reply:
had a second thought: maybe using as a default `name = "formula"` would make most sense, but I would give the user the possibility to choose which variable/column he/she wants as `rownames`.

Reply:
Sounds very reasonable :+1:
Yes, `name = "formula"` as default makes sense for most applications. `name = "compound_id"` would be nice if users want to add the adduct m/z columns to the compDb or if they want to index them later with the id.

By the way, after adding `...` in the generic, I've noticed that the `mass2mz` inside the method no longer defaults to the `MetaboCoreUtils` definition, and instead it tries to dispatch an S4 method. Since there's no method defined for a `numeric` input (cmps$exactmass), this raises an error. 
Adding the `MetaboCoreUtils::` prefix to the inner `mass2mz` call bypasses the problem. Is that the correct way to fix it?

Reply:
that is strange, but not unexpected. Having mixed functions and methods could indeed be tricky. So, yes, your proposed solution is OK.